client/my-sites/sidebar/sidebar.jsx

```diff
@@ -345,13 +346,25 @@ export class MySitesSidebar extends Component {
 	};
 
 	store() {
-		const { canUserManageOptions, isJetpack, site, siteSuffix, translate } = this.props;
+		// IMPORTANT: If you add a country to this list, you must also add it
+		// to ../../extensions/woocommerce/lib/countries in the getCountries function
+		const allowedCountryCodes = [ 'US', 'CA' ];
+		const { currentUser, canUserManageOptions, isJetpack, site, siteSuffix, translate } = this.props;
 		const storeLink = '/store' + siteSuffix;
 		const showStoreLink = config.isEnabled( 'woocommerce/extension-dashboard' ) &&
 			site && isJetpack && canUserManageOptions && this.props.isSiteAutomatedTransfer;
 
+		if ( ! showStoreLink ) {
```

Comment:
I totally agree that we should return ASAP (given how often this is re-rendered), but how about renaming this variable? Its name has become a bit confusing with these changes - it does control if the store link is shown, but at the same time, it's not the _only_ flag that controls it (`isCountryAllowed` being the second one, but that's not obvious). Especially since `isCountryAllowed` is the one used "near" the store link.
Maybe something like `isSiteEligible` would work better?